src/test/java/org/opensearch/neuralsearch/query/HybridQueryInnerHitsIT.java

```diff
@@ -0,0 +1,582 @@
+        assertEquals((double) searchHit1.get("_score"), (double) explanationForHit1.get("value"), DELTA_FOR_SCORE_ASSERTION);
+
+        // top level explanation
+        String expectedTopLevelDescription = "arithmetic_mean combination of:";
+        assertEquals(expectedTopLevelDescription, explanationForHit1.get("description"));
+
+        // Normalization explanation
+        List<Map<String, Object>> hit1Details = getListOfValues(explanationForHit1, "details");
+        assertEquals(1, hit1Details.size());
+        Map<String, Object> hit1DetailsForHit1 = hit1Details.get(0);
+        assertEquals(0.001f, (double) hit1DetailsForHit1.get("value"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals("min_max normalization of:", hit1DetailsForHit1.get("description"));
+        assertEquals(1, ((List) hit1DetailsForHit1.get("details")).size());
+
+        // Combination explanation
+        List<Map<String, Object>> hit1CombinationDetails = getListOfValues(hit1DetailsForHit1, "details");
+        assertEquals(1, hit1CombinationDetails.size());
+        Map<String, Object> internalCombinationDetailsForHit1 = hit1CombinationDetails.get(0);
+        assertEquals(
+            scoreOfInnerHits.get(TEST_NESTED_FIELD_NAME_1).get(0),
+            (double) internalCombinationDetailsForHit1.get("value"),
+            DELTA_FOR_SCORE_ASSERTION
+        );
+        assertEquals("combined score of:", internalCombinationDetailsForHit1.get("description"));
+        assertEquals(2, ((List) internalCombinationDetailsForHit1.get("details")).size());
+
+        // InnerHitsChild explanation
+        List<Map<String, Object>> hit1ChildDetails = getListOfValues(internalCombinationDetailsForHit1, "details");
+        assertEquals(2, hit1ChildDetails.size());
+
+        Map<String, Object> child1Details = hit1ChildDetails.get(0);
+        assertEquals(scoreOfInnerHits.get(TEST_NESTED_FIELD_NAME_1).get(0), (double) child1Details.get("value"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals(
+            "Score based on "
+                + innerHitCountPerFieldName.get(TEST_NESTED_FIELD_NAME_1).get(0)
+                + " child docs in range from 0 to 11, using score mode Max",
+            child1Details.get("description")
+        );
+        assertEquals(1, ((List) child1Details.get("details")).size());
+
+        Map<String, Object> child2Details = hit1ChildDetails.get(1);
+        assertEquals(scoreOfInnerHits.get(TEST_NESTED_FIELD_NAME_2).get(0), (double) child2Details.get("value"), DELTA_FOR_SCORE_ASSERTION);
+        assertEquals(
+            "Score based on "
+                + innerHitCountPerFieldName.get(TEST_NESTED_FIELD_NAME_2).get(0)
+                + " child docs in range from 0 to 11, using score mode Max",
+            child2Details.get("description")
+        );
+        assertEquals(1, ((List) child2Details.get("details")).size());
+    }
+
+    @SneakyThrows
+    private void initializeIndexIfNotExist(String indexName) {
```

Comment:
Are we doing cleanup of the indices created?

Reply:
Yeah it is in BaseNeuralSearchIT. Someone from heemin's team did that change.